src/main/java/baseball/controller/GameManager.java

```diff
@@ -0,0 +1,62 @@
+package baseball.controller;
+
+import baseball.io.PlayerIoManager;
+import baseball.model.ball.Ball;
+import baseball.model.ball.BallMaker;
+import baseball.model.judge.BaseballJudge;
+import baseball.model.judge.JudgeResult;
+import java.util.List;
+
+public class GameManager {
+
+    private PlayerIoManager playerIoManager;
+    private BallMaker ballMaker;
+    private BaseballJudge baseballJudge;
+
+    private GameManager() {
+    }
+
+    public static GameManager create() {
+        GameManager gameManager = new GameManager();
+
+        gameManager.playerIoManager = PlayerIoManager.create();
+        gameManager.ballMaker = new BallMaker();
+
+        return gameManager;
+    }
```

Comment:
이렇게 개발하게 되면 GameManager 객체를 생성할 때마다, PlayerIoManager와 BallMaker 객체를 새로 생성하게 되는데, 사실 그럴 필요없이 두 객체는 하나의 객체를 두고 새로운 GameManager객체 생성 시 DI를 통해 재활용할 수 있을 것이라고 생각합니다!
메모리 낭비가 사라지는거죠! 아래는 한번 변경해본 코드입니다! 참고해보시고 의견부탁드립니다!

```suggestion
    private final PlayerIoManager playerIoManager;
    private final BallMaker ballMaker;
    
    private BaseballJudge baseballJudge;

    private GameManager(PlayerIoManager playerIoManager, BallMaker ballMaker) {
        this.playerIoManager = playerIoManager;
        this.ballMaker = ballMaker;
    }

    public static GameManager from(PlayerIoManager playerIoManager, BallMaker ballMaker) {
        GameManager gameManager = new GameManager(playerIoManager, ballMaker);
    }
```